tools/fuzzer/expr_gen.h

```diff
@@ -40,26 +41,65 @@ enum class ExprKind : unsigned char {
 };
 constexpr size_t NUM_GEN_EXPR_KINDS = (size_t)ExprKind::EnumLast + 1;
 
+struct ExprKindWeightInfo {
+  float initial_weight;
+  float dampening_factor;
+};
+
+using BinOpMask = std::bitset<NUM_BIN_OPS>;
+using UnOpMask = std::bitset<NUM_UN_OPS>;
+
+struct GenConfig {
+  int num_exprs_to_generate = 20;
+
+  uint64_t int_const_min = 0;
+  uint64_t int_const_max = 1000;
+
+  double double_constant_min = 0;
+  double double_constant_max = 10;
+
+  float parenthesize_prob = 0.5f;
+
+  BinOpMask bin_op_mask = ~0ull ^ ((1ull << (unsigned long long)BinOp::Shl) |
```

Comment:
Can you do something like this?

```
BinOpMask().flip()
           .flip(BinOp::Shl)
           ...
           .flip(BinOp::Shr);
```

Idk if it will look nice after `clang-format` tho.